lib/elixir/test/elixir/kernel/lexical_tracker_test.exs

```diff
@@ -149,33 +149,89 @@ defmodule Kernel.LexicalTrackerTest do
           NotAModule
           Remote.func()
           R.func()
+          &extract/2
+          &is_record/1
+          &R.func/0
+          &Remote.func/0
+          &Integer.is_even/1
         end
 
+        &extract/2
+        &is_record/1
+        &R.func/0
+        &Remote.func/0
+        &Integer.is_even/1
+
         Kernel.LexicalTracker.remote_dispatches(__ENV__.module)
       end |> elem(3)
       """)
 
-    assert compile_remote_calls == %{
-      Bitwise => %{{:&&&, 2} => [9]},
-      Integer => %{{:is_even, 1} => [9]},
-      Kernel => %{{:and, 2} => [8]},
-      Kernel.LexicalTracker => %{{:remote_dispatches, 1} => [16]},
-      :elixir_def => %{{:store_definition, 6} => [6]}
-    }
-
-    assert runtime_remote_calls == %{
-      Record => %{{:extract, 2} => [7]},
-      Remote => %{{:func, 0} => [13, 12]},
-      :erlang => %{
-        {:==, 2} => [9],
-        {:>, 2} => [8],
-        {:andalso, 2} => [8],
-        {:band, 2} => [9],
-        {:element, 2} => [8],
-        {:is_atom, 1} => [8],
-        {:is_tuple, 1} => [8],
-        {:tuple_size, 1} => [8]
-      }
-    }
+    assert unroll_dispatches(compile_remote_calls) == [
```

Comment:
Moving this to a list is great but I am afraid of it being too fragile. If we change how `Kernel.def` works internally, it will break. That said, what if we make it a list, but check only the values we care? Something like:

```elixir
compile_remote_calls = unroll_dispatches(compile_remote_calls)
assert {6, Kernel, :def, 2} in compile_remote_calls
assert {8, Record, :is_record, 1} in compile_remote_calls
```

Thoughts?